packages/create-vite/template-preact-ts/package.json

```diff
@@ -5,7 +5,7 @@
   "type": "module",
   "scripts": {
     "dev": "vite",
-    "build": "tsc && vite build",
+    "build": "tsc -b && vite build",
```

Comment:
This generates `vite.config.js` and `vite.config.d.ts`. So we cannot merge this.

Normally adding `noEmit: true` to `tsconfig.node.json` works but in this case, we cannot simply set that. TypeScript doesn't allow it (`Referenced project 'D:/documents/GitHub/vite/packages/create-vite/template-preact-ts/tsconfig.node.json' may not disable emit.` error happens) (related: https://github.com/microsoft/TypeScript/issues/53979).

We can make it work by adding a tsconfig that references both `tsconfig.app.json` and `tsconfig.node.json` like create-vue does (https://github.com/vuejs/create-vue/pull/274). It seems there's a small problem with that though (https://github.com/vuejs/create-vue/issues/437).

Reply:
Thanks for pointing that out @sapphi-red, I hadn't realised that it was generating `vite.config.js` and `vite.config.d.ts` :sweat_smile: 

I've updated the PR to do it exactly how its done in https://github.com/vuejs/create-vue/blob/f75fd9813a624b8e44b012608335721901aba00b/template/tsconfig/base/tsconfig.json. Testing locally looks promising - I temporarily added a simple TS error to both a src file and `vite.config.json` and the errors in both show up fine, and only once.

I don't think https://github.com/vuejs/create-vue/issues/437 will be an issue here, from reading that issue it sounds like it happens because the two `tsconfig.*.json` files included the same set of files, whereas here there is no overlap - `tsconfig.app.json` includes only `src` while `tsconfig.node.json` only includes `vite.config.json`.

If you're happy with this, would you be able to let me know what you would like me to do, if anything, with the `svelte-ts` template? I'm unfamiliar with `svelte-check` so I don't know if its even possible to pass `-b` through.

Reply:
Just pushed an update to set `tsBuildInfoFile` to `node_modules/.tmp/` as its done in create-vue to avoid cluttering up the project root with various `.tsbuildinfo` files, see: https://github.com/vuejs/create-vue/blob/main/template/tsconfig/base/tsconfig.node.json#L13

Reply:
> If you're happy with this, would you be able to let me know what you would like me to do, if anything, with the svelte-ts template?

It seems `svelte-check` doesn't support project references. For now, I think running `svelte-check && tsc -p tsconfig.node.json` instead would be fine.

> I don't think vuejs/create-vue#437 will be an issue here.

On second thought, I think this is fine. As long as it's not deduped by TypeScript, we won't be able to solve this. Running two `tsc` (one for app and one for node) alongside would output the error in each, too.

Reply:
@sapphi-red 

> It seems svelte-check doesn't support project references. For now, I think running svelte-check && tsc -p tsconfig.node.json instead would be fine.

I've pushed the follwing update:

```diff
-    "check": "svelte-check --tsconfig ./tsconfig.json"
+    "check": "svelte-check --tsconfig ./tsconfig.json && tsc -p tsconfig.node.json"
```

> On second thought, I think this is fine. As long as it's not deduped by TypeScript, we won't be able to solve this. Running two tsc (one for app and one for node) alongside would output the error in each, too.

Makes sense :+1: